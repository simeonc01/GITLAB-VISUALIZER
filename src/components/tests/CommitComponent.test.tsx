// import renderer from "react-test-renderer";

// const testData = [
//     {
//         "title": "Merge branch '27-fix-typos-in-types-ts' into 'main'",
//         "author": "Victor Windsor Torbjørn Norris",
//         "commited_date": "26/09",
//         "message": "Merge branch '27-fix-typos-in-types-ts' into 'main'\n\nrefactor(#27): fix the typos in types.ts\n\nCloses #27\n\nSee merge request it2810-h22/Team-29/prosjekt2!20"
//     },
//     {
//         "title": "refactor(#27): fix the typos in types.ts",
//         "author": "bjerkemj",
//         "commited_date": "26/09",
//         "message": "refactor(#27): fix the typos in types.ts\n"
//     },
//     {
//         "title": "Merge branch '5-create-simple-design-prototype' into 'main'",
//         "author": "Victor Windsor Torbjørn Norris",
//         "commited_date": "19/09",
//         "message": "Merge branch '5-create-simple-design-prototype' into 'main'\n\nResolve \"create simple design/prototype\"\n\nCloses #10\n\nSee merge request it2810-h22/Team-29/prosjekt2!19"
//     },
//     {
//         "title": "feature(#5): properly displays projectname",
//         "author": "bjerkemj",
//         "commited_date": "19/09",
//         "message": "feature(#5): properly displays projectname\n"
//     },
//     {
//         "title": "Add package-lock to .gitignore",
//         "author": "bjerkemj",
//         "commited_date": "19/09",
//         "message": "Add package-lock to .gitignore\n"
//     }
// ]

// it("Only component test", () => {
//   const comp = renderer.create(<CommitTable rows ={[]} />).toJSON();
//   expect(comp).toMatchSnapshot();
// });

// it("Test with data", () => {
//     const comp = renderer.create(<CommitTable rows ={testData} />).toJSON();
//     expect(comp).toMatchSnapshot();
//   });
