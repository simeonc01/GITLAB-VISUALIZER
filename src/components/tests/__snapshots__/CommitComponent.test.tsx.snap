// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Only component test 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-k4y317-MuiTypography-root"
  >
    Commit log
  </h6>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1uzl8r8-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="Commits-table"
      className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          />
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Title
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Author
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-i02g0k-MuiTableCell-root"
            scope="col"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;

exports[`Test with data 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-k4y317-MuiTypography-root"
  >
    Commit log
  </h6>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1uzl8r8-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="Commits-table"
      className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          />
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Title
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
            scope="col"
          >
            Author
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-i02g0k-MuiTableCell-root"
            scope="col"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              aria-label="expand row"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </button>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Merge branch '27-fix-typos-in-types-ts' into 'main'
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Victor Windsor Torbj√∏rn Norris
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            26/09
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colSpan={6}
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          />
        </tr>
        <tr
          className="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              aria-label="expand row"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </button>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            refactor(#27): fix the typos in types.ts
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            bjerkemj
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            26/09
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colSpan={6}
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          />
        </tr>
        <tr
          className="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              aria-label="expand row"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </button>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Merge branch '5-create-simple-design-prototype' into 'main'
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Victor Windsor Torbj√∏rn Norris
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            19/09
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colSpan={6}
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          />
        </tr>
        <tr
          className="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              aria-label="expand row"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </button>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            feature(#5): properly displays projectname
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            bjerkemj
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            19/09
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colSpan={6}
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          />
        </tr>
        <tr
          className="MuiTableRow-root css-2blr9s-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              aria-label="expand row"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </button>
          </td>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Add package-lock to .gitignore
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            bjerkemj
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            19/09
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colSpan={6}
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
